arch/risc-v/src/mpfs/mpfs_serial.c

```diff
@@ -1547,6 +1554,14 @@ int up_putc(int ch)
 #ifdef HAVE_SERIAL_CONSOLE
   struct up_dev_s *priv = (struct up_dev_s *)CONSOLE_DEV.priv;
   uint32_t ier;
+#endif
+
+  if (!CONSOLE_DEV.isconsole)
```

Comment:
this is probably not right, CONSOLE_DEV should be inside HAVE_SERIAL_CONSOLE ifdef I suppose